src/com/hillel/homework/lesson15/TemperatureConversionMain.java

```diff
@@ -0,0 +1,66 @@
+package com.hillel.homework.lesson15;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class TemperatureConversionMain {
+
+    enum TemperatureScale {
+        KELVIN,
+        FAHRENHEIT
+    }
+
+    private static final BufferedReader READER = new BufferedReader(new InputStreamReader(System.in));
+
+    public static void main(String[] args) throws IOException {
+
+        double celsius = getCelsiusTemperature();
+        TemperatureScale scale = getConversionType();
+        convertTemperature(scale, celsius);
+    }
+
+    private static double getCelsiusTemperature() {
+        try {
+            System.out.print("Enter temperature in Celsius: ");
+            return Double.parseDouble(READER.readLine());
+        } catch (Exception exception) {
+            System.out.println("Please enter a number. Error: " + exception.getMessage());
+            return getCelsiusTemperature();
+        }
+    }
+
+    private static TemperatureScale getConversionType() {
+        try {
+            System.out.println("Please enter a conversion type (1 - Kelvin, 2 - Fahrenheit): ");
+            int choice = Integer.parseInt(READER.readLine());
+            switch (choice) {
+                case 1:
+                    return TemperatureScale.KELVIN;
+                case 2:
+                    return TemperatureScale.FAHRENHEIT;
+                default:
+                    System.out.println("Invalid input. Please enter 1 or 2.");
+                    return getConversionType();
+            }
+        } catch (Exception exception) {
+            System.out.println("Please enter a number 1 or 2. Error: " + exception.getMessage());
+            return getConversionType();
+        }
+    }
+
+    private static void convertTemperature(TemperatureScale scale, double celsius) {
```

Comment:
this code may be improved.
you ca split it to 2 steps:
1. create Converter and return it
2. call covert method just once for the converter

Reply:
I see, thanks for the tip! I've made the changes in a new commit called 'refactor: convertTemperature method'.

Please take a look and let me know if you have any questions.